scripts/train_lm.py

```diff
@@ -196,19 +196,17 @@ def FormatMetaparameters(metaparameters, num_train_sets):
 
     return ','.join(out_param)
 
-# this function checks whether there are repeating values in parameters(excluding 
-# the first #number numbers, which are not data-type weights) and if found, return
-# the index of the first pair repeating values 
+# this function checks repeating values in parameters and returns the index
+# of the first pair of repeating values if any is found 
 def FindRepeatingValue(parameters):
-#assert (number >= 1) and (number < len(parameters))
     for i in range(0, len(parameters)):
         for j in range(i + 1, len(parameters)):
             if parameters[i] == parameters[j]:
                 return(i, j)
     return (-1, -1)
 
-# this function rerounds repeating values until they are different and return the values after rerounding
-# if they are indeed the same (collison), exit with a warning (this case can be very rare) 
+# this function rerounds repeating values until they are different and returns the values after rerounding.
+# if they are indeed the same (collison), it exits with a warning (this case can be very rare) 
 def AdjustValues(number1, number2, index1, index2, ref_number1, ref_number2):
```

Comment:
I think if two values are the same (collision), it would be better to just print them with full precision rather than die.
The reason is that collisions are only disallowed in certain circumstances:  for the same order of discounting parameter, we cannot have d1 == d2 or d2 == d3.  But otherwise, collisions are quite possible.  Making this cause the script to die is a bad idea.


And we cannot have two data-directory weights be identical [IIRC].  But otherwise, collisions